### PR TITLE
test: add coverage for invalid parameters for `rescanblockchain`

### PR DESCRIPTION
**Pull request by brunoerg**
This PR adds test coverage for the following errors:
https://github.com/bitcoin/bitcoin/blob/2bd9aa5a44b88c866c4d98f8a7bf7154049cba31/src/wallet/rpc/transactions.cpp#L880-L894

### CODE REVIEW

**Review by aureleoules — COMMENTED** [review]
ACK 32cd1441d7e178b14ff63154585eaf0f38104cae - i verified this case was not tested.

Maybe outside the scope of this PR but this also seems to be untested.
https://github.com/bitcoin/bitcoin/blob/2bd9aa5a44b88c866c4d98f8a7bf7154049cba31/src/wallet/rpc/transactions.cpp#L896-L899

**Review by w0xlt — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/25906/commits/32cd1441d7e178b14ff63154585eaf0f38104cae

**Comment by brunoerg** [comment]
@aureleoules 

> Maybe outside the scope of this PR but this also seems to be untested.

Cool, I can work on it for other PR. Thanks

File: test/functional/wallet_transactiontime_rescan.py [hunk 1]
```diff
@@ -158,5 +159,10 @@ def run_test(self):
                 assert_equal(tx['time'], cur_time + ten_days + ten_days + ten_days)
```
Comment (kristapsk) [inline test/functional/wallet_transactiontime_rescan.py:161]: Add some log message here? For example:
```suggestion
        self.log.info('Test handling of invalid parameters for rescanblockchain')
```

Comment (brunoerg) [reply test/functional/wallet_transactiontime_rescan.py:161]: Make sense, done!

**Comment by brunoerg** [comment]
Force pushed addressing @kristapsk's review.

**Review by kristapsk — APPROVED** [review]
ACK bd47bb56681b0cc0b0c49e9b4e7f3d6e2ae5ad3e

**Review by w0xlt — APPROVED** [review]
reACK https://github.com/bitcoin/bitcoin/pull/25906/commits/bd47bb56681b0cc0b0c49e9b4e7f3d6e2ae5ad3e

But there is a lint error in CI.

**Comment by brunoerg** [comment]
> But there is a lint error in CI.

Thanks, @w0xlt. It was fixed in last force-pushed.

**Review by w0xlt — APPROVED** [review]
reACK https://github.com/bitcoin/bitcoin/pull/25906/commits/d1a00046214c02684438adcfcd23eea39b86bc7f

CI error in [Win64 native [vs2022]](https://cirrus-ci.com/task/6646678206808064) check seems unrelated.

**Pull request merged**